### DISCUSSION TITLE
Layout Component Remounting On Route Change

### DISCUSSION THREAD

**Discussion by fweth** [facebook/docusaurus#10322] [General] [answered]
I noticed that the Layout component remounts on route change (I checked via `React.useEffect(() => console.log("remount"), [])`).  Is this supposed to be that way?  I wanted to have some kind of responsive, animated logo in the header which should not be interrupted from route changes.

**Comment by slorber** [comment, marked as answer]
Yes we consider this to be a bug tracked in https://github.com/facebook/docusaurus/issues/2891

Unfortunately, the layout has to be applied on a per-page basis because we need to allow users to use multiple layouts, or no layout at all. Fixing it will likely result in implementing a new feature with a public API.
